zulip_bots/zulip_bots/bots/idonethis/idonethis.py

```diff
@@ -0,0 +1,212 @@
+import sys
+import requests
+import logging
+import re
+
+from typing import Any, Dict
+
+API_BASE_URL = "https://beta.idonethis.com/api/v2"
+
+api_key = ""
+default_team = ""
+
+class AuthenticationException(Exception):
+    pass
+
+class TeamNotFoundException(Exception):
+    def __init__(self, team: str) -> None:
+        self.team = team
+
+class UnknownCommandSyntax(Exception):
+    def __init__(self, detail: str) -> None:
+        self.detail = detail
+    pass
+
+class UnspecifiedProblemExcept(Exception):
+    pass
+
+def makeAPIRequest(endpoint: str, method: str="GET", body: Dict[str, str]=None) -> Any:
+    global api_key
+    headers = {'Authorization': 'Token ' + api_key}
+    if method == "GET":
+        r = requests.get(API_BASE_URL + endpoint, headers=headers)
+    elif method == "POST":
+        r = requests.post(API_BASE_URL + endpoint, headers=headers, json=body)
+    if r.status_code == 200:
+        return r.json()
+    elif r.status_code == 401 and 'error' in r.json() and r.json()['error']  == "Invalid API Authentication":
+        logging.error('Error authenticating, please check key ' + str(r.url))
+        raise AuthenticationException()
+    else:
+        logging.error('Error make API request, code ' + str(r.status_code) + '. json: ' + r.json())
+        raise UnspecifiedProblemExcept()
+
+def apiNoop() -> Any:
+    return makeAPIRequest("/noop")
+
+def apiListTeam() -> Any:
```

Comment:
The Python community generally uses "snake_case" casing for functions, so I would rename functions to be like this:

* api_list_team
* api_show_team
* api_show_users
* api_list_entries